helper/fairshare/jobmanager.go

```diff
@@ -364,3 +363,14 @@ func (j *JobManager) removeLastQueueAccessed() {
 		j.lastQueueAccessed = len(j.queuesIndex) - 1
 	}
 }
+
+// Pause adds the ability to temporarily stop the job manager. It can be unpaused using Unpause.
+// It is a no-op if the job manager is already paused.
+func (j *JobManager) Pause() {
```

Comment:
is there any reason to have pause be sync so that calling code can know when current in-progress work is complete?